src/librustc_driver/lib.rs

```diff
@@ -360,21 +361,71 @@ fn handle_explain(code: &str,
     };
     match descriptions.find_description(&normalised) {
         Some(ref description) => {
-            // Slice off the leading newline and print.
-            print!("{}", &(&description[1..]).split("\n").map(|x| {
+            let output = &(&description[1..]).split("\n").map(|x| {
                 format!("{}\n", if x.starts_with("```") {
                     "```"
                 } else {
                     x
                 })
-            }).collect::<String>());
+            }).collect::<String>();
+
+            show_content_with_pager(output);
+
         }
         None => {
             early_error(output, &format!("no extended information for {}", code));
         }
     }
 }
 
+#[cfg(not(windows))]
+fn spawn_pager_with_file(file_path: &PathBuf) -> io::Result<Output> {
+    use std::os::unix::io::FromRawFd;
+    let raw_file_descriptor_id = 2; // stdout
+
+    let pager = env::var("PAGER").unwrap_or(String::from("less"));
```

Comment:
Since it's quite easy, could we use `var_os` here (and in the companion windows function)?